crates/extension/src/extension_store.rs

```diff
@@ -243,10 +243,7 @@ impl ExtensionStore {
             extension_index: Default::default(),
             installed_dir,
             index_path,
-            builder: Arc::new(ExtensionBuilder::new(
-                http::client(http_client.proxy().cloned()),
-                build_dir,
-            )),
+            builder: Arc::new(ExtensionBuilder::new(build_dir, http_client.clone())),
```

Comment:
why don't we just clone http_client here?

Reply:
Because we don't want to use the fake HTTP client in the tests.

Reply:
Okay, let me change that back.